tests/python_api_tests.py

```diff
@@ -1,51 +1,420 @@
 # coding=utf-8

```

Comment:
Should we convert this to python tests instead of running it serially?

Pros:
- a failure doesn't prevent the other tests from running.
- report of which test cases failed.
- easily run a single test case without having to comment out code.

Reply:
That did occur to me, and I agree we should do it. Especially now, with it being so much longer. That's part of my long-term plan for updating the project, but I was hoping to put it off while I work on the `kapigen` conversion.

Added http://b/358191743

Reply:
+1 to converting it, and also to figuring out some way to mock out the server, since there's a fair amount of code here that's related to handling server-side delays.

I think it would be OK to defer these changes, but I'm concerned that there's no automated validation in the script.  It just prints out responses, and the developer has to know if the response is expected or not.  That's hard for new folks to understand, and even for the expert, it's error prone.  It would be great to update this script at least so that it has some automated checks and a `PASS` or `FAIL` message for each test.  Wdyt?